### PR TITLE
Add Azure Terraform modules

### PR DESCRIPTION
**Pull request by diogocp**
Initial version of the Azure modules for testing in staging.

### CODE REVIEW

File: modules/azure/custom-data/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+## Modules
+
+No modules.
+
+## Resources
+
+| Name | Type |
+|------|------|
+| [terraform_data.template](https://registry.terraform.io/providers/hashicorp/terraform/latest/docs/resources/data) | resource |
```
Comment (drichards-87) [inline modules/azure/custom-data/README.md:22]: Don't know if this is expected, but the link returns a Page Not Found error.

File: modules/azure/custom-data/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| <a name="input_api_key"></a> [api\_key](#input\_api\_key) | Specifies the API key required by the Datadog Agent to submit vulnerabilities to Datadog | `string` | n/a | yes |
```
Comment (drichards-87) [inline modules/azure/custom-data/README.md:28]:
```suggestion
| <a name="input_api_key"></a> [api\_key](#input\_api\_key) | Specifies the API key required by the Datadog Agent to submit vulnerabilities to Datadog. | `string` | n/a | yes |
```

File: modules/azure/custom-data/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner resources will be created | `string` | n/a | yes |
```
Comment (drichards-87) [inline modules/azure/custom-data/README.md:29]:
```suggestion
| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner resources will be created. | `string` | n/a | yes |
```

File: modules/azure/custom-data/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+| <a name="input_site"></a> [site](#input\_site) | By default the Agent sends its data to Datadog US site. If your organization is on another site, you must update it. See https://docs.datadoghq.com/getting_started/site/ | `string` | `"datadoghq.com"` | no |
```
Comment (drichards-87) [inline modules/azure/custom-data/README.md:30]:
```suggestion
| <a name="input_site"></a> [site](#input\_site) | By default, the Agent sends its data to the Datadog US site. If your organization is on another site, you must update it. See https://docs.datadoghq.com/getting_started/site/ for more information. | `string` | `"datadoghq.com"` | no |
```

File: modules/azure/custom-data/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## Outputs
+
+| Name | Description |
+|------|-------------|
+| <a name="output_agent_version"></a> [agent\_version](#output\_agent\_version) | The version of the Datadog Agent installed |
```
Comment (drichards-87) [inline modules/azure/custom-data/README.md:36]:
```suggestion
| <a name="output_agent_version"></a> [agent\_version](#output\_agent\_version) | The version of the Datadog Agent installed. |
```

File: modules/azure/custom-data/README.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+| <a name="output_install_sh"></a> [install\_sh](#output\_install\_sh) | The installation script for the Datadog agentless scanner |
```
Comment (drichards-87) [inline modules/azure/custom-data/README.md:37]:
```suggestion
| <a name="output_install_sh"></a> [install\_sh](#output\_install\_sh) | The installation script for the Datadog Agentless Scanner. |
```

File: modules/azure/custom-data/README.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+| <a name="output_scanner_version"></a> [scanner\_version](#output\_scanner\_version) | The version of the Datadog Agentless Scanner installed |
```
Comment (drichards-87) [inline modules/azure/custom-data/README.md:38]:
```suggestion
| <a name="output_scanner_version"></a> [scanner\_version](#output\_scanner\_version) | The version of the Datadog Agentless Scanner installed. |
```

File: modules/azure/managed-identity/README.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+|------|------|
+| [azurerm_role_assignment.role_assignment](https://registry.terraform.io/providers/hashicorp/azurerm/3.101.0/docs/resources/role_assignment) | resource |
+| [azurerm_role_definition.role_definition](https://registry.terraform.io/providers/hashicorp/azurerm/3.101.0/docs/resources/role_definition) | resource |
+| [azurerm_user_assigned_identity.managed_identity](https://registry.terraform.io/providers/hashicorp/azurerm/3.101.0/docs/resources/user_assigned_identity) | resource |
+
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| <a name="input_location"></a> [location](#input\_location) | The location where the Datadog Agentless Scanner resources will be created | `string` | n/a | yes |
```
Comment (drichards-87) [inline modules/azure/managed-identity/README.md:31]:
```suggestion
| <a name="input_location"></a> [location](#input\_location) | The location where the Datadog Agentless Scanner resources will be created. | `string` | n/a | yes |
```

File: modules/azure/managed-identity/README.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+| <a name="input_name"></a> [name](#input\_name) | Name of the managed identity to be assigned to the Datadog Agentless Scanner virtual machine instances | `string` | `"DatatogAgentlessScanner"` | no |
```
Comment (drichards-87) [inline modules/azure/managed-identity/README.md:32]:
```suggestion
| <a name="input_name"></a> [name](#input\_name) | Name of the managed identity to be assigned to the Datadog Agentless Scanner virtual machine instances. | `string` | `"DatatogAgentlessScanner"` | no |
```

File: modules/azure/managed-identity/README.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+| <a name="input_resource_group_id"></a> [resource\_group\_id](#input\_resource\_group\_id) | The ID of the resource group to which the role assignment will be scoped | `string` | n/a | yes |
```
Comment (drichards-87) [inline modules/azure/managed-identity/README.md:33]:
```suggestion
| <a name="input_resource_group_id"></a> [resource\_group\_id](#input\_resource\_group\_id) | The ID of the resource group to which the role assignment will be scoped. | `string` | n/a | yes |
```

File: modules/azure/managed-identity/README.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+| <a name="input_resource_group_name"></a> [resource\_group\_name](#input\_resource\_group\_name) | The name of the resource group where the managed identity will be created | `string` | n/a | yes |
+| <a name="input_tags"></a> [tags](#input\_tags) | A map of additional tags to add to the managed identity | `map(string)` | `{}` | no |
```
Comment (drichards-87) [inline modules/azure/managed-identity/README.md:35]:
```suggestion
| <a name="input_tags"></a> [tags](#input\_tags) | A map of additional tags to add to the managed identity. | `map(string)` | `{}` | no |
```

File: modules/azure/managed-identity/README.md [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+
+## Outputs
+
+| Name | Description |
+|------|-------------|
+| <a name="output_identity"></a> [identity](#output\_identity) | The managed identity to be assigned to the Datadog Agentless Scanner virtual machine instances |
```
Comment (drichards-87) [inline modules/azure/managed-identity/README.md:41]:
```suggestion
| <a name="output_identity"></a> [identity](#output\_identity) | The managed identity to be assigned to the Datadog Agentless Scanner virtual machine instances. |
```

File: modules/azure/resource-group/README.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+| Name | Type |
+|------|------|
+| [azurerm_resource_group.resource_group](https://registry.terraform.io/providers/hashicorp/azurerm/3.101.0/docs/resources/resource_group) | resource |
+
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner resources will be created | `string` | n/a | yes |
+| <a name="input_name"></a> [name](#input\_name) | The name of the resource group where the Datadog Agentless Scanner resources will be created | `string` | `"DatatogAgentlessScanner"` | no |
```
Comment (drichards-87) [inline modules/azure/resource-group/README.md:30]:
```suggestion
| <a name="input_name"></a> [name](#input\_name) | The name of the resource group where the Datadog Agentless Scanner resources will be created. | `string` | `"DatatogAgentlessScanner"` | no |
```

File: modules/azure/virtual-machine/README.md [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
+| <a name="input_name"></a> [name](#input\_name) | Name prefix to be used on EC2 instance created | `string` | `"DatadogAgentlessScanner"` | no |
+| <a name="input_resource_group_name"></a> [resource\_group\_name](#input\_resource\_group\_name) | The name of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
+| <a name="input_subnet_id"></a> [subnet\_id](#input\_subnet\_id) | The ID of the subnet to launch in | `string` | n/a | yes |
+| <a name="input_tags"></a> [tags](#input\_tags) | A map of additional tags to add to the instance/volume created | `map(string)` | `{}` | no |
+| <a name="input_user_assigned_identity"></a> [user\_assigned\_identity](#input\_user\_assigned\_identity) | The resource ID of the managed identity to be assigned to the Datadog Agentless Scanner virtual machine instances | `string` | n/a | yes |
```
Comment (drichards-87) [inline modules/azure/virtual-machine/README.md:40]:
```suggestion
| <a name="input_user_assigned_identity"></a> [user\_assigned\_identity](#input\_user\_assigned\_identity) | The resource ID of the managed identity to be assigned to the Datadog Agentless Scanner virtual machine instances. | `string` | n/a | yes |
```

File: modules/azure/virtual-machine/README.md [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+| <a name="input_instance_root_volume_size"></a> [instance\_root\_volume\_size](#input\_instance\_root\_volume\_size) | The instance root volume size in GiB | `number` | `30` | no |
+| <a name="input_instance_size"></a> [instance\_size](#input\_instance\_size) | The type of instance | `string` | `"Standard_B2ps_v2"` | no |
+| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
+| <a name="input_name"></a> [name](#input\_name) | Name prefix to be used on EC2 instance created | `string` | `"DatadogAgentlessScanner"` | no |
+| <a name="input_resource_group_name"></a> [resource\_group\_name](#input\_resource\_group\_name) | The name of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
+| <a name="input_subnet_id"></a> [subnet\_id](#input\_subnet\_id) | The ID of the subnet to launch in | `string` | n/a | yes |
+| <a name="input_tags"></a> [tags](#input\_tags) | A map of additional tags to add to the instance/volume created | `map(string)` | `{}` | no |
```
Comment (drichards-87) [inline modules/azure/virtual-machine/README.md:39]:
```suggestion
| <a name="input_tags"></a> [tags](#input\_tags) | A map of additional tags to add to the instance/volume created. | `map(string)` | `{}` | no |
```

File: modules/azure/virtual-machine/README.md [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+| <a name="input_instance_root_volume_size"></a> [instance\_root\_volume\_size](#input\_instance\_root\_volume\_size) | The instance root volume size in GiB | `number` | `30` | no |
+| <a name="input_instance_size"></a> [instance\_size](#input\_instance\_size) | The type of instance | `string` | `"Standard_B2ps_v2"` | no |
+| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
+| <a name="input_name"></a> [name](#input\_name) | Name prefix to be used on EC2 instance created | `string` | `"DatadogAgentlessScanner"` | no |
+| <a name="input_resource_group_name"></a> [resource\_group\_name](#input\_resource\_group\_name) | The name of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
+| <a name="input_subnet_id"></a> [subnet\_id](#input\_subnet\_id) | The ID of the subnet to launch in | `string` | n/a | yes |
```
Comment (drichards-87) [inline modules/azure/virtual-machine/README.md:38]:
```suggestion
| <a name="input_subnet_id"></a> [subnet\_id](#input\_subnet\_id) | The ID of the subnet to launch in. | `string` | n/a | yes |
```

File: modules/azure/virtual-machine/README.md [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+| <a name="input_instance_count"></a> [instance\_count](#input\_instance\_count) | Size of the autoscaling group the instance is in (i.e. number of instances to run) | `number` | `1` | no |
+| <a name="input_instance_root_volume_size"></a> [instance\_root\_volume\_size](#input\_instance\_root\_volume\_size) | The instance root volume size in GiB | `number` | `30` | no |
+| <a name="input_instance_size"></a> [instance\_size](#input\_instance\_size) | The type of instance | `string` | `"Standard_B2ps_v2"` | no |
+| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
+| <a name="input_name"></a> [name](#input\_name) | Name prefix to be used on EC2 instance created | `string` | `"DatadogAgentlessScanner"` | no |
+| <a name="input_resource_group_name"></a> [resource\_group\_name](#input\_resource\_group\_name) | The name of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
```
Comment (drichards-87) [inline modules/azure/virtual-machine/README.md:37]:
```suggestion
| <a name="input_resource_group_name"></a> [resource\_group\_name](#input\_resource\_group\_name) | The name of the resource group where the Datadog Agentless Scanner network resources will be created. | `string` | n/a | yes |
```

File: modules/azure/virtual-machine/README.md [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+| <a name="input_admin_username"></a> [admin\_username](#input\_admin\_username) | Name of the admin user to use for the instance | `string` | `"azureuser"` | no |
+| <a name="input_custom_data"></a> [custom\_data](#input\_custom\_data) | The user data to provide when launching the instance | `string` | `null` | no |
+| <a name="input_instance_count"></a> [instance\_count](#input\_instance\_count) | Size of the autoscaling group the instance is in (i.e. number of instances to run) | `number` | `1` | no |
+| <a name="input_instance_root_volume_size"></a> [instance\_root\_volume\_size](#input\_instance\_root\_volume\_size) | The instance root volume size in GiB | `number` | `30` | no |
+| <a name="input_instance_size"></a> [instance\_size](#input\_instance\_size) | The type of instance | `string` | `"Standard_B2ps_v2"` | no |
+| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
+| <a name="input_name"></a> [name](#input\_name) | Name prefix to be used on EC2 instance created | `string` | `"DatadogAgentlessScanner"` | no |
```
Comment (drichards-87) [inline modules/azure/virtual-machine/README.md:36]:
```suggestion
| <a name="input_name"></a> [name](#input\_name) | Name prefix to be used on the EC2 instance created. | `string` | `"DatadogAgentlessScanner"` | no |
```

File: modules/azure/virtual-machine/README.md [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+| <a name="input_admin_ssh_key"></a> [admin\_ssh\_key](#input\_admin\_ssh\_key) | SSH public key for the admin user | `string` | n/a | yes |
+| <a name="input_admin_username"></a> [admin\_username](#input\_admin\_username) | Name of the admin user to use for the instance | `string` | `"azureuser"` | no |
+| <a name="input_custom_data"></a> [custom\_data](#input\_custom\_data) | The user data to provide when launching the instance | `string` | `null` | no |
+| <a name="input_instance_count"></a> [instance\_count](#input\_instance\_count) | Size of the autoscaling group the instance is in (i.e. number of instances to run) | `number` | `1` | no |
+| <a name="input_instance_root_volume_size"></a> [instance\_root\_volume\_size](#input\_instance\_root\_volume\_size) | The instance root volume size in GiB | `number` | `30` | no |
+| <a name="input_instance_size"></a> [instance\_size](#input\_instance\_size) | The type of instance | `string` | `"Standard_B2ps_v2"` | no |
+| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner network resources will be created | `string` | n/a | yes |
```
Comment (drichards-87) [inline modules/azure/virtual-machine/README.md:35]:
```suggestion
| <a name="input_location"></a> [location](#input\_location) | The location of the resource group where the Datadog Agentless Scanner network resources will be created. | `string` | n/a | yes |
```

File: modules/azure/virtual-machine/README.md [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+
+## Inputs
+
+| Name | Description | Type | Default | Required |
+|------|-------------|------|---------|:--------:|
+| <a name="input_admin_ssh_key"></a> [admin\_ssh\_key](#input\_admin\_ssh\_key) | SSH public key for the admin user | `string` | n/a | yes |
+| <a name="input_admin_username"></a> [admin\_username](#input\_admin\_username) | Name of the admin user to use for the instance | `string` | `"azureuser"` | no |
+| <a name="input_custom_data"></a> [custom\_data](#input\_custom\_data) | The user data to provide when launching the instance | `string` | `null` | no |
+| <a name="input_instance_count"></a> [instance\_count](#input\_instance\_count) | Size of the autoscaling group the instance is in (i.e. number of instances to run) | `number` | `1` | no |
+| <a name="input_instance_root_volume_size"></a> [instance\_root\_volume\_size](#input\_instance\_root\_volume\_size) | The instance root volume size in GiB | `number` | `30` | no |
```
Comment (drichards-87) [inline modules/azure/virtual-machine/README.md:33]:
```suggestion
| <a name="input_instance_root_volume_size"></a> [instance\_root\_volume\_size](#input\_instance\_root\_volume\_size) | The instance root volume size in GiB. | `number` | `30` | no |
```

File: modules/azure/virtual-machine/README.md [hunk 21]
```diff
@@ -0,0 +1,45 @@
...
+| <a name="input_instance_size"></a> [instance\_size](#input\_instance\_size) | The type of instance | `string` | `"Standard_B2ps_v2"` | no |
```
Comment (drichards-87) [inline modules/azure/virtual-machine/README.md:34]:
```suggestion
| <a name="input_instance_size"></a> [instance\_size](#input\_instance\_size) | The type of instance. | `string` | `"Standard_B2ps_v2"` | no |
```

**Review by drichards-87 — DISMISSED** [review]
Left some feedback from Docs and approved the PR.

**Comment by diogocp** [comment]
I think it's better to make the documentation changes in another PR, because we should make the same changes in the AWS modules.

File: modules/azure/virtual-machine/main.tf [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+
+  identity {
+    type = "UserAssigned"
+    identity_ids = [
+      var.user_assigned_identity
+    ]
+  }
+  os_profile {
+    linux_configuration {
+      computer_name_prefix            = "dd-agentless-"
```
Comment (jinroh) [inline modules/azure/virtual-machine/main.tf:26]: We prefix them with `agentless-scanning-*` on AWS. Would it make sense to be consistent here ?

File: modules/azure/virtual-machine/main.tf [hunk 23]
```diff
@@ -0,0 +1,56 @@
+locals {
+  dd_tags = {
+    Datadog                 = "true"
+    DatadogAgentlessScanner = "true"
+  }
+}
+
+resource "azurerm_orchestrated_virtual_machine_scale_set" "vmss" {
```
Comment (jinroh) [inline modules/azure/virtual-machine/main.tf:8]: Is there any ability to automatically restart VMs on a fixed period ? We do that for ASGs, instances are recycled every 24 hours.

It helps with auto-updating.

File: modules/azure/virtual-machine/main.tf (hunk 22 above)
Comment (diogocp) [reply modules/azure/virtual-machine/main.tf:26]: I think I ran into a string length limitation, but I can double-check.

Comment (diogocp) [reply modules/azure/virtual-machine/main.tf]: Done.

File: modules/azure/virtual-machine/main.tf (hunk 23 above)
Comment (diogocp) [reply modules/azure/virtual-machine/main.tf:8]: I will leave this for later.

Comment (diogocp) [reply modules/azure/virtual-machine/main.tf:8]: I found [Start/Stop VMs v2](https://learn.microsoft.com/en-us/azure/azure-functions/start-stop-vms/overview) but it seems a bit overkill.

**Pull request merged**